internal/mempool/mempool.go

```diff
@@ -1149,14 +1192,21 @@ func (mp *TxPool) maybeAcceptTransaction(tx *dcrutil.Tx, isNew, rateLimit, allow
 	// Determine what type of transaction we're dealing with (regular or stake).
 	// Then, be sure to set the tx tree correctly as it's possible a use submitted
 	// it to the network with TxTreeUnknown.
-	txType := stake.DetermineTxType(msgTx)
+	txType := stake.DetermineTxType(msgTx, isTreasuryEnabled)
 	if txType == stake.TxTypeRegular {
 		tx.SetTree(wire.TxTreeRegular)
 	} else {
 		tx.SetTree(wire.TxTreeStake)
 	}
 	isVote := txType == stake.TxTypeSSGen
 
+	var isTreasuryBase, isTSpend bool
+	if isTreasuryEnabled {
+		// XXX do we need TADD here?
+		isTSpend = txType == stake.TxTypeTSpend
+		isTreasuryBase = txType == stake.TxTypeTreasuryBase
+	}
+
```

Comment:
On line 1238 (below this point) there's the following check: 
```go
	// Reject votes before stake validation height.
	stakeValidationHeight := mp.cfg.ChainParams.StakeValidationHeight
	if isVote && nextBlockHeight < stakeValidationHeight {
		str := fmt.Sprintf("votes are not valid until block height %d (next "+
			"block height %d)", stakeValidationHeight, nextBlockHeight)
		return nil, txRuleError(ErrInvalid, str)
	}
```

This should probably be amended to also reject TSpends before SVH.

Reply:
Fixed.